src/internal.jl

```diff
@@ -0,0 +1,20 @@
+
+import UUIDs
+
+"""Internal storage common to PowerSystems types."""
+struct PowerSystemInternal
+    uuid::Base.UUID
+end
+
+"""Creates PowerSystemInternal with a UUID."""
+PowerSystemInternal() = PowerSystemInternal(UUIDs.uuid4())
+
+"""Gets the UUID for any PowerSystemType."""
+function get_uuid(obj::T)::Base.UUID where T <: PowerSystemType
```

Comment:
This can be `get_uuid(obj::PowerSystemType)` and does not need the `where T <: PowerSystemType` as we discussed.

Reply:
Fixed